### PR TITLE
feat: ConnectUI settings backend

### PR DESCRIPTION
**Pull request by kaposke**
> DB Migration PR #4574 

- `GET /api/v1/connect-ui-settings` - Returns connectUI settings for environment. Fallback to default settings (always returns something)
- `PUT /api/v1/connect-ui-settings` - Upserts new settings
- (Update) `GET /api/v1/connect/session` - Adds connectUI settings to output payload (this is how connectUI will get the theme and watermark flag)

---

**Environment-Specific Connect UI Settings Backend & API Integration**

This PR introduces backend support for environment-specific Connect UI settings, providing REST API endpoints to retrieve and update settings per environment. It includes database schema changes, type and service layer additions, validation logic, and integrates Connect UI settings into the connect session API payload. Comprehensive integration and contract tests are provided, and types are exposed via shared code for frontend consumption. The feature allows per-environment customization of the Connect UI's appearance (theme colors and watermark), with gated access controlled by plan definitions.

<details>
<summary><strong>Key Changes</strong></summary>

• Introduced ```GET`` /api/v1/connect-ui-settings` and ```PUT`` /api/v1/connect-ui-settings` endpoints for retrieving/upserting environment-specific Connect ``UI`` settings with validation and error handling.
• Added new service methods for connect ``UI`` settings (``CRUD``) and a shared ``defaultConnectUISettings`` constant.
• Created a new ``connect_ui_settings`` database table and integrated the feature into the migration and plan definition system for feature-gating.
• Updated the `/connect/session` ``API`` to include connect ``UI`` settings in its output payload.
• Exposed settings and contract types in shared code for both backend and frontend usage.
• Added frontend utility (`setTheme`) for applying backend-provided theme settings (currently only light mode).
• Wrote comprehensive integration and contract tests for various edge cases (protection, validation, multi-env correctness).

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• ``API``: /api/v1/connect-ui-settings (``GET``, ``PUT``) endpoints
• Connect session controller and schema
• Database schema (migrations, new table: `connect_ui_settings`)
• Backend services and shared type system
• Frontend theme utility
• Integration and contract testing suites

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/connect-ui/src/lib/theme.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { ConnectUIThemeSettings } from '@nangohq/types';
+
+export function setTheme(theme: ConnectUIThemeSettings) {
+    const root = document.documentElement;
+
+    console.log(root.style);
```
Comment (bodinsamuel) [inline packages/connect-ui/src/lib/theme.ts]:
```suggestion
```

File: packages/shared/lib/services/connect-ui-settings.service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            .insert({
+                environment_id: environmentId,
+                theme: settings.theme,
+                show_watermark: settings.showWatermark
+            })
+            .onConflict('environment_id')
+            .merge();
+        return Ok(undefined);
+    } catch (err) {
+        console.error(err);
```
Comment (bodinsamuel) [inline packages/shared/lib/services/connect-ui-settings.service.ts]:
```suggestion
```

File: packages/server/lib/controllers/v1/connectUISettings/putConnectUISettings.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import db from '@nangohq/database';
+import { connectUISettingsService } from '@nangohq/shared';
+import { report, requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+
+import type { PutConnectUISettings } from '@nangohq/types';
+
+const colorPaletteSchema = z
+    .object({
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/v1/connectUISettings/putConnectUISettings.ts]: preferred way in v4
```suggestion
    .strictObject({
```

**Review by bodinsamuel — APPROVED** [review]
